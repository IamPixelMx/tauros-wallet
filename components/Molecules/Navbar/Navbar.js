import React, { useState } from 'react';

import { NavbarItem } from '.';

const NAV_ITEMS = [
  { page: 'Tarjeta Tauros', route: '/' },
  { page: 'Wallets', route: '/' },
  { page: 'Exchange', route: '/' },
];
const DROP_ITEMS = [{ label: 'Configuración' }, { label: 'Cerrar Sesión' }];

const Navbar = props => {
  const [isNavOpen, setIsNavOpen] = useState(false);
  const [isDropOpen, setIsDropOpen] = useState(false);

  const toggleNav = () => {
    setIsNavOpen(!isNavOpen);
  };

  const toggleDrop = () => {
    setIsDropOpen(!isDropOpen);
  };

  return (
    <nav className='navbar'>
      <div className='container'>
        <div className='navbar-brand'>
          <figure className='image is-128x128'>
            <img src='/tauros.png' alt='logo' />
          </figure>
          <span
            className={isNavOpen ? 'navbar-burger is-active' : 'navbar-burger'}
            data-target='navbarMenuHeroB'
            onClick={toggleNav}
          >
            <span></span>
            <span></span>
            <span></span>
          </span>
        </div>
        <div className={isNavOpen ? 'navbar-menu is-active' : 'navbar-menu'} id='navbarMenuHeroB'>
          <div className={isNavOpen ? 'navbar-end is-active' : 'navbar-end'}>
            {NAV_ITEMS.map(item => (
              <NavbarItem key={`${item.page}-nav-item`} {...item} {...props} />
            ))}

            <div
              className={
                isDropOpen ? 'navbar-item has-dropdown is-active' : 'navbar-item has-dropdown'
              }
              onClick={() => {
                toggleDrop();
                toggleNav();
              }}
            >
              <hr className='navbar-divider' />
              <a className='navbar-link is-arrowless'>Perfil</a>
              <div className='navbar-dropdown is-boxed is-right'>
                {DROP_ITEMS.map(({ label }) => (
                  <a key={`${label}-drop`} className='navbar-item'>
                    {label}
                  </a>
                ))}
              </div>
            </div>
          </div>
        </div>
      </div>
    </nav>
  );
};

export default Navbar;
