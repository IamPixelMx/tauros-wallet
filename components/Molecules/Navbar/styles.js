import styled from '@emotion/styled';
import { css } from '@emotion/core';

export const hasSize = ({ size }) =>
  size === 'small' &&
  css`
    height: calc(65px - (var(--spacing-l) * 2));
  `;

export const NavbarStyled = styled.div`
  line-height: var(--font-size-m);
  ${hasSize}
`;
