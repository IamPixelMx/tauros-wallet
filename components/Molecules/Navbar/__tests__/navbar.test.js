import React from 'react';
import serializer from 'jest-emotion';

import { renderWithProviderSnapshot } from 'jest-utils';
import Navbar from 'components';

expect.addSnapshotSerializer(serializer);

describe('Navbar', () => {
  const navbarElement = renderWithProviderSnapshot(<Navbar />).toJSON();

  it('Renders navbar', () => {
    expect(navbarElement).toMatchSnapshot();
  });
});
