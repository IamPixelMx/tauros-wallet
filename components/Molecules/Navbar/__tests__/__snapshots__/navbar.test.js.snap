// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Renders Navbar 1`] = `
<nav
  className="navbar"
>
  <div
    className="container"
  >
    <div
      className="navbar-brand"
    >
      <figure
        className="image is-128x128"
      >
        <img
          alt="logo"
          src="/tauros.png"
        />
      </figure>
      <span
        className="navbar-burger"
        data-target="navbarMenuHeroB"
        onClick={[Function]}
      >
        <span />
        <span />
        <span />
      </span>
    </div>
    <div
      className="navbar-menu"
      id="navbarMenuHeroB"
    >
      <div
        className="navbar-end"
      >
        <a
          className="navbar-item is-nice-blue"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Tarjeta Tauros
        </a>
        <a
          className="navbar-item is-nice-blue"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Wallets
        </a>
        <a
          className="navbar-item is-nice-blue"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Exchange
        </a>
        <a
          className="navbar-item is-nice-blue"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Perfil
        </a>
      </div>
    </div>
  </div>
</nav>
`;
